etcdserver/api/v3rpc/grpc.go

```diff
@@ -36,8 +36,11 @@ func Server(s *etcdserver.EtcdServer, tls *tls.Config) *grpc.Server {
 	}
 	opts = append(opts, grpc.UnaryInterceptor(newUnaryInterceptor(s)))
 	opts = append(opts, grpc.StreamInterceptor(newStreamInterceptor(s)))
-
+	// set MaxMsgSize be a bit higher than that of MaxRequestBytes bypasses grpc request size check
```

Comment:
I noticed that grpc does an internal check of the incoming request. if the incoming request is greater than MaxMsgSize, it returns a grpc.internal error to client immediately. thus, the request never goes to etcd server and this check will never be performed,
```go
if len(data) > maxRequestBytes {
		return nil, ErrRequestTooLarge
}
```
https://github.com/coreos/etcd/blob/master/etcdserver/v3_server.go#L609.

which breaks some of our tests that expect `ErrRequestTooLarge`.

Just realized that adding 1024 won't work. If the request size is > (MaxRequestBytes + 1024),  then grpc will return an error to client immediately.

I need to rework on handling `ErrRequestTooLarge`. It seems me there is no way to bypass the grpc request size check completely and let the etcd server to do the check for us.